src/main/java/reactor/netty/transport/ProxyProvider.java

```diff
@@ -324,6 +340,87 @@ public ProxyProvider build() {
 		}
 	}
 
+	static final class RegexShouldProxyPredicate implements Predicate<SocketAddress> {
+
+		public static final RegexShouldProxyPredicate ALWAYS_PROXY = RegexShouldProxyPredicate.fromWildcardedPattern("");
+		public static final RegexShouldProxyPredicate DEFAULT_NON_PROXY = RegexShouldProxyPredicate.fromWildcardedPattern("localhost|127.*|[::1]|0.0.0.0|[::0]");
+
+		private final String regex;
+		private final Pattern pattern;
+
+		/**
+		 * Create a {@link RegexShouldProxyPredicate} based off the provided regular expression.
+		 *
+		 * @param regex The string regular expression
+		 * @return a predicate whether we should bypass the proxy
+		 */
+		public static RegexShouldProxyPredicate fromRegularExpression(String regex) {
```

Comment:
Do we need this, it is not used at the moment

Reply:
I did it for completeness since I made the constructor private and I figured I should support to existing way of defining a normal java pattern to represent nonProxyHosts. I am happy to remove though.

Reply:
yes please remove it